python/revng/api/_capi.py

```diff
@@ -0,0 +1,44 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+from pathlib import Path
+
+from cffi import FFI
+from cffi.backend_ctypes import CTypesBackend
+
+from .utils import make_c_string
+
+ORCHESTRA_ROOT = Path(__file__).parent.parent.parent.parent.parent
+HEADERS_DIR = ORCHESTRA_ROOT / "include" / "revng" / "PipelineC"
+
+header_paths = [
+    HEADERS_DIR / "c_typedefs.h",
+    HEADERS_DIR / "prototypes.h",
+]
+
+ffi = FFI()
+for header_path in header_paths:
+    with open(header_path) as header_file:
+        lines = []
+        for line in header_file:
+            if not line.startswith("#"):
+                lines.append(line)
+        ffi.cdef("\n".join(lines))
+
+LIB_DIR = ORCHESTRA_ROOT / "lib"
+LIBRARY_PATH = str(LIB_DIR / "librevngPipelineC.so")
+
+ARGV = []
+ANALYSIS_LIBRARIES = [
+    str(LIB_DIR / "revng/analyses/librevngFunctionIsolation.so"),
+]
+
+_ANALYSIS_LIBRARIES = [make_c_string(s) for s in ANALYSIS_LIBRARIES]
+
+
+ctypes_backend = CTypesBackend()
+
+_api = ffi.dlopen(LIBRARY_PATH, flags=ctypes_backend.RTLD_NOW | ctypes_backend.RTLD_GLOBAL)
+if not _api.rp_is_initialized():
```

Comment:
I'm not sure why we need this.
I think that we should assert that `rp_initialize` is called only once in its body.

Reply:
That would mean passing ownership of the `Initialized` variable to python, I don't think it's a good idea